.buildkite/scripts/steps/openapi_publishing/publish_oas_docs.sh

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+source .buildkite/scripts/common/util.sh
+
+echo "--- Publish OAS docs"
+
+echo "--- Installing NPM modules"
+
+deploy_to_bump() {
+  local file_path="${1:-}"
+  local doc_name="${2:-}"
+  local doc_token="${3:-}"
+  local branch="${4:-}"
+  local cur_dir=$(pwd)
+
+  cd "$(dirname "${BASH_SOURCE[0]}")"
+  npm install
+
+  echo "Checking diff for doc '$doc_name' against file '$file_path' on branch '$branch'..."
+  local result=$(npx bump diff $file_path --doc $doc_name --token $doc_token --branch $branch --format=json)
+  local change_count=$(jq '. | length' <<<$result)
+  if [[ ! -z $change_count && $change_count -gt 0 ]]; then
+    echo "Found $change_count changes..."
+    echo "About to deploy file '$file_path' to doc '$doc_name' to '$branch' on bump.sh..."
+    npx bump deploy $file_path \
+      --branch $branch \
+      --doc $doc_name \
+      --token $doc_token ;
+    echo ""
+    echo "Note: if there is a warning of unchanged docs we probably have unpublished deployments waiting."
+    echo "Go to https://bump.sh/elastic/dashboard to see all the docs in the hub."
+  else
+    echo "Did not detect changes for '$file_path'; not deploying. Got response: '$result'"
+  fi
+
+  echo "Switch back to dir '$cur_dir'"
+  cd $cur_dir
+}
+
+if [[ "$BUILDKITE_BRANCH" == "main" ]]; then
+  BUMP_KIBANA_DOC_NAME="$(vault_get kibana-bump-sh kibana-doc-name)"
+  BUMP_KIBANA_DOC_TOKEN="$(vault_get kibana-bump-sh kibana-token)"
+  deploy_to_bump "$(pwd)/oas_docs/output/kibana.yaml" $BUMP_KIBANA_DOC_NAME $BUMP_KIBANA_DOC_TOKEN main;
+  exit 0;
+fi
+
+if [[ "$BUILDKITE_BRANCH" == "8.x" ]]; then
```

Comment:
> if [[ "$BUILDKITE_BRANCH" == "8.x" ]]; then

I presume in this branch we'll need to add another clause like this, but for the 8.16 branch and targeting the "v8" branch in Bump.sh

Reply:
Yeah, that's a good point... this step won't do anything on this branch.

One alternative for back port branches: we could "set and forget" to e.g. 8.16. But I guess that would mean a lot more branches on bump. Perhaps worth leaving out this back port for now. Wdyt?

Reply:
Since it looks like we have at least one more 8.16 release before 8.17 takes over, I'm leaning toward having the clause in here to publish to the v8 branch. The docs team just needs to remember to comment it out when it comes time to switch to 8.17.

Reply:
I've added an 8.16 section in d90c609872b9fa8ebdbdc41174e859538d152f67

Reply:
++ that sounds good, we'll have to change the check to `8.17` on that branch